smartsim/_core/_install/buildenv.py

```diff
@@ -58,24 +61,28 @@ class SetupError(Exception):
     See setup.py for more
     """
 
-    pass
-
 
 # so as to not conflict with pkg_resources.packaging.version.Version
+# pylint: disable-next=invalid-name
 class Version_(str):
     """A subclass of pkg_resources.packaging.version.Version that
     includes some helper methods for comparing versions.
     """
 
-    def _convert_to_version(self, vers: t.Union[str, Iterable[packaging.version.Version], packaging.version.Version]) -> t.Any:
+    @staticmethod
```

Comment:
This one would be better as a classmethod so you can use the class parameter in the isinstance checks